rust/kernel/error.rs

```diff
@@ -352,22 +352,42 @@ impl Error {
     pub fn to_kernel_errno(self) -> c_types::c_int {
         self.0
     }
+
+    /// Returns a string representing the error, if one exists.
+    #[cfg(not(testlib))]
+    pub fn name(&self) -> Option<&'static CStr> {
+        // SAFETY: Just an FFI call, there are no extra safety requirements.
+        let ptr = unsafe { bindings::errname(-self.0) };
+        if ptr.is_null() {
+            None
+        } else {
+            // SAFETY: The string returned by `errname` is static and `NUL`-terminated.
+            Some(unsafe { CStr::from_char_ptr(ptr) })
+        }
+    }
+
+    /// Returns a string representing the error, if one exists.
+    ///
+    /// When `testlib` is configured, this always returns `None` to avoid the dependency on a
+    /// kernel function so that tests that use this (e.g., by calling [`Result::unwrap`]) can still
+    /// run in userspace.
+    #[cfg(testlib)]
+    pub fn name(&self) -> Option<&'static CStr> {
+        None
+    }
 }
 
 impl fmt::Debug for Error {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        // SAFETY: FFI call.
-        let name = unsafe { bindings::errname(-self.0) };
-
-        if name.is_null() {
+        let name = if let Some(n) = self.name() {
```

Comment:
Perhaps
```rust
match self.name() {
    Some(name) => {
        // SAFETY: These strings are ASCII-only.
        let str = unsafe { str::from_utf8_unchecked(name) };
        f.debug_tuple(str).finish()
    }
    None => {
        // Print out number if no name can be found.
        f.debug_tuple("Error").field(&-self.0).finish()
    }
}
```

Reply:
Sure, I changed to code to use a `match`.